Improve kernel structure to be the main repo from CC

miguelinux opened the pull request:
To set packaging as main repo some improvements were made.

- Now, update_kernel.sh fetch kernel version from upstream
- Keeps a release track version
- Deleted files not needed files in repo
- Updates CC mailing list address

miguelinux committed:
kernel: update spec file with CDN URL

* Use CDN URL to download the kernel from upstream.
* Clean some BuildRequires lines, as they are already required by devel
  package.
* Remove target from BuildKernel function, the target all is the default
  one used by make.

Fixes #26

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>

gorozco1 commented on kernel/update_kernel.sh:
```diff
@@ -6,26 +6,48 @@
 AUTHOR=${AUTHOR:-$(git config user.name)}
 AUTHOR_EMAIL=${AUTHOR_EMAIL:-$(git config user.email)}
 
-CC_VERSIONS_FILE="../versions.txt"
```
why ignoring ../versions.txt?

miguelinux replied:
The idea is that packaging/kernel will be the first place where the kernel will be updated and after build test and a PR to packaging  will be propagated to others repos

gorozco1 commented on kernel/update_kernel.sh:
```diff
@@ -6,26 +6,48 @@
-CC_VERSIONS_FILE="../versions.txt"
-source "$CC_VERSIONS_FILE"
-clear_vm_kernel_version=$(echo $clear_vm_kernel_version | cut -d'-' -f1)
-VERSION=${1:-$clear_vm_kernel_version}
```
we need to be able to build "any" version of the kernel with parameter

miguelinux replied:
Ok, I'll fix it

gorozco1 reviewed (changes requested):
please clarify

miguelinux commented:
@gorozco1 changes are integrated

gorozco1 commented on kernel/update_kernel.sh:
```diff
@@ -2,30 +2,62 @@
+    if [ ! -f ${KFILE} ]
+    then
+        echo "Downloading linux kernel from ${KSOURCE}"
+        if ! curl --fail -O ${KSOURCE}
```
Please remove the conditional for "file" and always try the download of the kernel with the continue flag.


`curl -L -O -C - "${KSOURCE}"`

miguelinux commented:
@gorozco1 the change is done, Could you review? please.

MarioCarrilloA commented on kernel/update_kernel.sh:
```diff
@@ -96,4 +76,36 @@ then
+
+ln -sf linux-$VERSION.tar.xz linux-container_$VERSION.orig.tar.xz
+
+for i in `cat obs_working_dir/debian/patches/series`
```
Maybe you could use $() instead of ``

MarioCarrilloA commented:
@miguelinux could you please update README file too :)

miguelinux committed:
gitignore: add Vim tmp swap file and generated files

* Add the swap temp file created by Vim when editing text.

* Add the generated files from update_kernel.sh

Fixes #26

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>

miguelinux committed:
kernel: improve a little the update kernel script

In this commit the following changes were done

* Remove the debug flag to only output message, you can run this script
  with "$ bash -x update_kernel.sh" to enable debug output.

* Set/Add URLs to download the release.json and sha256sums.asc in a
  variable.

* The script now fetch the latest LTS kernel version number from upstream,
  but you can pass by parameter the kernel version you want to build.

* A release file was included to track the release number.

* Add brackets to VERSION variable.

Fixes #26

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>

miguelinux committed:
kernel: update debian copyright linux kernel site

The URL to download the Linux kernel was updated taking in consideration
the following information from https://www.kernel.org/category/faq.html

What happened to ftp.kernel.org?

FTP service was terminated on March 1, 2017. All content that used
to be available via ftp.kernel.org can be accessed by browsing
https://www.kernel.org/pub/. If you would like to use a
command-line tool for accessing these files, you can do so with lftp:

    lftp https://www.kernel.org/pub

Also the emails were updated to clear continers team.

Fixes #26

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>

miguelinux committed:
kernel: update README.md with the new update sequence

Update the README file with the update sequence.

Fixes #26

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>

miguelinux commented:
@gorozco1, @MarioCarrilloA changes done
